lms/services/roster.py

```diff
@@ -316,6 +326,127 @@ def fetch_canvas_group_roster(self, canvas_group: LMSSegment) -> None:
             update_columns=["active", "updated"],
         )
 
+    def fetch_canvas_sections_roster(self, lms_course: LMSCourse) -> None:
+        """Fetch the roster information for all canvas sections for one particular course.
+
+        Sections are different than other rosters:
+             - We fetch them via the propietary Canvas API, not the LTI Names and Roles endpoint.
+
+             - Due to the return value of that API we don't fetch rosters for indivual sections,
+               but for all sections of one course at once
+
+             - The return value of the API doesn't include enought information to create unseen users
+               so we'll only match against users we have seen before in the course.
+        """
+        application_instance = self._get_application_instance(lms_course)
```

Comment:
I'm using here single-purpose-helper-functions which I'm not generally a fan of but I reckon the help here with this long method.